src/MountedTraversal.js

```diff
@@ -65,6 +67,33 @@ export function instHasType(inst, type) {
   }
 }
 
+export function instHasAttribute(inst, compatAttrKey, attrValue) {
+  if (!isDOMComponent(inst)) return false;
+
+  // compatibility for html attributes or react props
+  const attrKey = attributePropMap[compatAttrKey] || compatAttrKey;
+  const node = getNode(inst);
+  let instProps;
+
+  /*
+   * we could gain from the `getAttribute` `HTMLElement` api, but
+   * for compatibility between `shallow` and `mount` We'll stick
+   * prop look up with a compatiblity check for the attr/react
+   * prop naming.
+   */
+  if (!node.props) { // React 0.13
+    instProps = node._store.props;
+  } else {
+    instProps = node.props;
+  }
```

Comment:
there should be a `propsOfNode()` function that abstracts this away... it's also important to check for falsy nodes as well where this would probably throw.